tools/add_polkit_action.py

```diff
@@ -152,23 +152,24 @@ def findDuplicateActions(self):
 
         ret = True
 
-        def checkDuplicate(entry):
-            if entry.action == self.m_args.action:
-                printerr("ERROR: action to be added already exists in {}:{}".format(
-                    entry.path, entry.linenr
-                ))
-                nonlocal ret
-                ret = False
-
-
         for profile in PROFILES:
 
             path = getProfilePath(profile)
 
-            parseProfile(path, checkDuplicate)
+            for entry in parseProfile(path):
+                ret &= self.checkDuplicate(entry)
```

Comment:
The `&=` is cute, but skimming over the code it made me stop for a minute to figure out what's going on. (Bitwise AND on a boolean).

Maybe just move the contents of `checkDuplicate` into the loop instead.

Reply:
I made a simple `if not` statement out of it.